settings.go

```diff
@@ -0,0 +1,224 @@
+	SyslinuxDirectory                 string                                `json:"syslinux_dir" mapstructure:"syslinux_dir"`
+	SyslinuxMemdiskDirectory          string                                `json:"syslinux_memdisk_folder" mapstructure:"syslinux_memdisk_folder"`
+	SyslinuxPxelinuxDirectory         string                                `json:"syslinux_pxelinux_folder" mapstructure:"syslinux_pxelinux_folder"`
+	TftpbootLocation                  string                                `json:"tftpboot_location" mapstructure:"tftpboot_location"`
+	VirtAutoBoot                      bool                                  `json:"virt_auto_boot" mapstructure:"virt_auto_boot"`
+	WebDirectory                      string                                `json:"webdir" mapstructure:"webdir"`
+	WebDirectoryWhitelist             []string                              `json:"webdir_whitelist" mapstructure:"webdir_whitelist"`
+	WindowsEnabled                    bool                                  `json:"windows_enabled" mapstructure:"windows_enabled"`
+	WindowsTemplateDirectory          string                                `json:"windows_template_dir" mapstructure:"windows_template_dir"`
+	XmlrpcPort                        int                                   `json:"xmlrpc_port" mapstructure:"xmlrpc_port"`
+	YumDistroPriority                 int                                   `json:"yum_distro_priority" mapstructure:"yum_distro_priority"`
+	YumDownloaderFlags                string                                `json:"yumdownloader_flags" mapstructure:"yumdownloader_flags"`
+	YumPostInstallMirror              bool                                  `json:"yum_post_install_mirror" mapstructure:"yum_post_install_mirror"`
+}
+
+// cobblerDataHacks is a hook for the mapstructure decoder. It's only used by
+// decodeCobblerItem and should never be invoked directly.
+// It's used to smooth out issues with converting fields and types from Cobbler.
+func cobblerSettingsHacks(f, t reflect.Kind, data interface{}) (interface{}, error) {
+	dataVal := reflect.ValueOf(data)
+
+	// Cobbler uses ~ internally to mean None/nil
+	if dataVal.String() == "~" {
+		return map[string]interface{}{}, nil
+	}
+
+	if f == reflect.Int64 && t == reflect.Bool {
+		if dataVal.Int() > 0 {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	}
+	return data, nil
+}
+
+// decodeCobblerItem is a custom mapstructure decoder to handler Cobbler's uniqueness.
+func decodeCobblerSettings(raw interface{}, result interface{}) (interface{}, error) {
+	var metadata mapstructure.Metadata
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
+		Metadata:         &metadata,
+		Result:           result,
+		WeaklyTypedInput: true,
+		DecodeHook:       cobblerSettingsHacks,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err := decoder.Decode(raw); err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
```

Comment:
Ideally we would deduplicate this with cobblerSignatureHacks in signatures.go, but for now I would be ok with just changing the f and t variable names